src/core/server/saved_objects/migrations/actions/reindex.ts

```diff
@@ -67,7 +67,7 @@ export const reindex =
             // Set reindex batch size
             size: BATCH_SIZE,
             // Exclude saved object types
-            query: unusedTypesQuery,
+            query: excludeOnUpgradeQuery,
```

Comment:
(outside of the scope of the PR) @rudolf Why do we need to still exclude those during the temp->target reindex, given the exclusion query was already used during the source->temp 'client-side' reindex?

https://github.com/elastic/kibana/blob/3864f63ccea1fe8aa27bc43fd9a6b9625cb19f6f/src/core/server/saved_objects/migrations/next.ts#L93-L104

Reply:
We use this during `REINDEX_SOURCE_TO_TEMP_READ` and `LEGACY_REINDEX`. Temp -> target is a clone (without any query)

The `LEGACY_REINDEX` probably doesn't benefit from this query because we don't have any bugs in < 7.3 which created thousands of objects.